pennylane/ops/functions/matrix.py

```diff
@@ -130,6 +130,8 @@ def cost(theta):
 @matrix.tape_transform
 def _matrix(tape, wire_order=None):
     """Defines how matrix works if applied to a tape containing multiple operations."""
+    if not tape.operations:
+        raise qml.operation.MatrixUndefinedError

```

Comment:
What if we only have measurements? I believe we want to return the identity then:

```python
>>> qscript = QuantumScript(measurements=[qml.expval(qml.PauliX(0))])
>>> qml.matrix(qscript)
array([[1., 0.],
       [0., 1.]])
```
Should we change this to `tape.wires` then?
```suggestion
    if not tape.wires:
        raise qml.operation.MatrixUndefinedError

```

Reply:
I can get behind that! I just picked operations because that's all we iterated over, but the default was Identity so we should keep that. maybe I'll add a test to ensure it too :)